field.go

```diff
@@ -219,16 +219,36 @@ func (f *Field) unpackVal(buf []byte, val reflect.Value, length int, options *Op
 		default:
 			return fmt.Errorf("struc: refusing to unpack float into field %s of type %s", f.Name, f.kind.String())
 		}
-	case Bool, Int8, Int16, Int32, Int64, Uint8, Uint16, Uint32, Uint64:
+	case Int8, Int16, Int32, Int64:
+		var n int64
+		switch typ {
```

Comment:
You can probably just put these switch cases in the unsigned `switch typ` below instead of duplicating the rest of the code.

Reply:
Done as of https://github.com/lunixbochs/struc/pull/58/commits/ce758f71082fbb85d7ab9e516b255045dd6b23e6